core/src/main/java/org/opensearch/sql/datasource/DataSourceServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.datasource;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.sql.datasource.model.DataSource;
+import org.opensearch.sql.datasource.model.DataSourceMetadata;
+import org.opensearch.sql.datasource.model.DataSourceType;
+import org.opensearch.sql.storage.DataSourceFactory;
+
+/**
+ * This class manages catalogs and responsible for creating connectors to these catalogs.
+ */
+public class DataSourceServiceImpl implements DataSourceService {
```

Comment:
No dependency on settings and it got out of plugin package[Which was odd earlier]. 

Earlier I had dependency on settings to construct opensearchStorageEngine and then after few refactorings I lost track that I don't need settings anymore.  

Now storageEngine construction is also done through factory class.